Add ChangeCondition function in parser

hoonori opened the pull request:
For universal use of multiple functions, parser can support managing condition change coming from options.

hoonori committed:
Added fail TC of ChangeCondition() for multiple functions

Going to use it for search, modify, delete functions

hoonori committed:
implemented ChangeCondition() functionality.

returns detailed KeyType if Option is active.

s33jo reviewed (approved):
LGTM, Search에 붙여서 테스트 진행해보도록 하겠습니다. 
궁금해서 찾아봤는데, 
조건이 3개일떄까지는 if가 빠르고 그이상은 switch가 더 빠르다고 하는군요..
정확하지는 않습니다.. 더 찾아보고 공유드릴게요

hoonori commented:
> LGTM, Search에 붙여서 테스트 진행해보도록 하겠습니다. 궁금해서 찾아봤는데, 조건이 3개일떄까지는 if가 빠르고 그이상은 switch가 더 빠르다고 하는군요.. 정확하지는 않습니다.. 더 찾아보고 공유드릴게요

넵:) 세명이 어떻게 구현했는지 다 읽어본 뒤에, 일단은 가장 가독성이 좋아보이는 것으로 가져왔습니다만, 다같이 여기서 대화하면서 최종적으로 정하는 것이 옳을 듯 합니다:)

제 현재 의견은, 전체 program 실행에서의 bottleneck은 옵션처리구간이 아닌 search, delete, modify operation 그 자체일 것으로 생각되어, 일단은 가독성을 좀 챙기면서 진행하는 것이 어떨까 했습니다. switch를 사용할 경우, switch 내부에 if가 한번 더 들어가야 하는 상태로 보여서요!

물론 제 예상이 틀릴 수도 있습니다! 다른 분들 의견은 어떠신가요?:)

noh0249 commented:
확인했습니다. 적용해보겠습니다.

s33jo commented:
> > LGTM, Search에 붙여서 테스트 진행해보도록 하겠습니다. 궁금해서 찾아봤는데, 조건이 3개일떄까지는 if가 빠르고 그이상은 switch가 더 빠르다고 하는군요.. 정확하지는 않습니다.. 더 찾아보고 공유드릴게요
> 
> 넵:) 세명이 어떻게 구현했는지 다 읽어본 뒤에, 일단은 가장 가독성이 좋아보이는 것으로 가져왔습니다만, 다같이 여기서 대화하면서 최종적으로 정하는 것이 옳을 듯 합니다:)
> 
> 제 현재 의견은, 전체 program 실행에서의 bottleneck은 옵션처리구간이 아닌 search, delete, modify operation 그 자체일 것으로 생각되어, 일단은 가독성을 좀 챙기면서 진행하는 것이 어떨까 했습니다. switch를 사용할 경우, switch 내부에 if가 한번 더 들어가야 하는 상태로 보여서요!
> 
> 물론 제 예상이 틀릴 수도 있습니다! 다른 분들 의견은 어떠신가요?:)

아 맞군요. switch안에 if문이 추가로 들어가 있군요!

hoonori committed:
Modified search, mod, del to use changecondition() function

to remove duplicate code, utilized new function supported by parser : ChangeCondition()

s33jo reviewed (approved):
LGTM, 테스트 해보겠습니다. search에서 -m, -l에서 last name, phone 넘버 구분한다고
isnumber() 함수 사용했는데, 삭제해도 되겠군요 감사합니다.
